src/main/java/com/duckblade/osrs/dpscalc/DpsCalcPlugin.java

```diff
@@ -117,6 +115,77 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 			navButton.getOnSelect().run();
 	}
 
+	// subscribe to equipment changes
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (!config.enableAutoSync()) return;
+
+		if (event.getContainerId() == InventoryID.EQUIPMENT.getId()) {
+			panel.onEquipmentChanged();
+		}
+	}
+
+	// subscribe to attack style changes
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		if (!config.enableAutoSync()) return;
+
+		if (event.getIndex() == VarPlayer.ATTACK_STYLE.getId()) {
+			panel.onEquipmentChanged();
```

Comment:
All these event handlers need to be queued and squashed between game ticks to prevent large swaps (e.g. a six-piece gear swap in one tick) causing six independent swing updates. Otherwise, this can cause a significant performance impact on the user's client.

As an alternative, this can use an `AtomicBoolean` toggled to true when the panel needs to be updated, and dispatch a swing thread panel update in `onGameTick`. This should also apply if you look into the overlay display.